zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js

```diff
@@ -87,16 +98,34 @@ export const getProfile = (refs, props) => {
  *
  * @todo - describe function's signature
  */
- export const deleteAccount = (username_el, props) => {
+export const deleteAccount = (username_el, props, toast) => {
+  let password_match = true;
   if (username_el.current.firstChild.value !== props.auth.username) {
-    return { dialogError: props.t('profile.delete.errors.incorrectUsernme') };
+    return { dialog_error: props.t('profile.delete.errors.incorrectUsername') };
+  } else if (!props?.values?.password)  {
+    return { dialog_error: props.t('profile.delete.errors.emptyPassword') };
   } else {
-    return props.deleteAccount({
-      token: props.auth.token,
-      history: props.history,
-      logout: props.logout,
-      t: props.t,
+    props.login({ values: props.values, history: props.history }).catch(error => {
```

Comment:
@kamthamc Is it ok to do this even after you are already authorized? I feel since the user is already authorized, this is more like a secondary check, but if you feel we can keep this without any harm, lets do it.

Reply:
I haven't seen any app asking to relogin. I only saw asking for a otp code sent to emails/sms.

Reply:
@chaitanya-cloudknox Let me give you some context:
 - The requirement here is that we need to ask for valid username and password before deleting the account - as this is a sensitive task. Currently it only asks for username
 - Given this, the approach @brrkrmn is taking is to ask for username and password and then do `props.login` again. Is that ok? Does this request the backend and do a thorough password/login challenge?

Reply:
Ok, I did some reading and I think this is fine for now